### ISSUE TITLE
Subscriptions via broadcastService not working for Angular wrapper

### ISSUE THREAD

**Issue by tipsanemonk** [AzureAD/microsoft-authentication-library-for-js#394]
There seems to be a problem with the loginRedirect option. The public APIs, as available in the MsalService has been utilized to login and getUser. The login reidrect works fine and gets back to the application with the token. But afterwards if we want to invoke getUser, it gives null and no boradcast subscriptions work as expected. Also, the MsalInterceptor gives error saying, <code>core.js:1671 ERROR user_login_error|User login is required</code>. It seems, the loginRedirect option is missing something. 

## The module is registered as following

<code>
MsalModule.forRoot({
  clientID: [my-client-id],
  consentScopes: ['user.read', 'User.Read.All']
})
</code>

## Interceptor is registered in the module providers list as: 
<code>
{
      provide: HTTP_INTERCEPTORS,
      useClass: MsalInterceptor,
      multi: true
   }
</code>

## Login in was done using the follwing 
<pre>
<code>
this.msalService.loginRedirect();
</code></pre>

## The subscriptions are called as below 

<pre>
<code>
this.broadcastService.subscribe('msal:loginSuccess', (payload) => {
      console.log(payload);
      });
</code>
</pre>

### Library version

<pre><code>
Library version: @azure/msal-angular 0.1.0
</code></pre>

## Current behavior
The callback is never exceuted and returns <code>null</code> user details on <code>msalService.getUser() </code>

## Expected behavior
The subscription callback is supposed to get invoked and return  user details on invoking msalService.getUser()

**Comment by nehaagrawal** [comment]
@tipsanemonk  I am not able to reproduce this issue. I am getting subscription  callback for loginRedirect use case. You can try our dev sample [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular/samples/MSALAngularDemoApp). Is it possible for you to share a repo where I can reproduce this? I have a feeling that you are missing the unsubscribe event in your component and this can cause this issue if you have multiple components listening to msal:loginSuccess. Some other component could be catching this event as well.

```
 ngOnDestroy() {
    this.broadcastService.getMSALSubject().next(1);
    if(this.subscription) {
      this.subscription.unsubscribe();
    }
  }
```

**Comment by tipsanemonk** [comment]
@nehaagrawal It seems the service only works with top-level routing components. In my case, I injected the service in a child route component and it wasn't working, but when I moved it to the app component, it worked fine.  

The initial login was a different child route and the redirected landing page is the different route which is again a child route.

Unfortunately, the repository I'm working on is private.
